### DISCUSSION TITLE
generic-manifest.p5m

### DISCUSSION THREAD

**Discussion by balrog-nona** [OpenIndiana/oi-userland#6390] [Q&A]
Hello,
I would like to ask about `generic-manifest.p5m` file. I've noticed this file in PR #6384 with a lot of content but the `sample-manifest.p5m` is also present.
I've found rules for generating this file in `make-rules/ips.mk` - these rules being added by #6154 
Does it mean that `generic-manifest.p5m` should be present in python components updated ever since?
Thank you!

**Comment by AndWac** [comment]
Am 11.01.21 um 14:45 schrieb Nona Hansel:
>
> Hello,
> I would like to ask about |generic-manifest.p5m| file. I've noticed
> this file in PR #6384
> <https://github.com/OpenIndiana/oi-userland/pull/6384> with a lot of
> content but the |sample-manifest.p5m| is also present.
> I've found rules for generating this file in |make-rules/ips.mk| -
> these rules being added by #6154
> <https://github.com/OpenIndiana/oi-userland/pull/6154>
> Does it mean that |generic-manifest.p5m| should be present in python
> components updated ever since?
> Thank you!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/OpenIndiana/oi-userland/discussions/6390>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADET4N2SZ4S2N2X3D3LYK43SZL6HPANCNFSM4V5SUM3A>.
>
Hi Nona,

yes, the generic-manifest.p5m files should be added to python
components. At least for those components that are available for more
than one python version. In the Makefile you typically have something
like PYTHON_VERSIONS= 3.5 3.7. 3.9 and a generic manifest file
<component name>-PYVER.p5m. The build machinery iterates over all
version that are listed in PYTHON_VERSIONS (note the plural -S at its end)
using the <component name>-PYVER.p5m file to generate a manifest for
each python version. The generic-manifest.p5m can be used as a
template for the <component name>-PYVER.p5m file and thus should also be
versioned.

Andreas

**Reply by alarcher** [reply]
Hi Nona,
as Andreas described it very well this generic manifest is a convenience template to update the <component name>-PYVER.p5m.
However the content delivered by a specific Python version may be different than others.
In this case you will need to annotate the manifest with macros to select/ignore lines based on the Python version.

This commit introduced the macros to include a line ONLY for a given version or EXCLude a line:

https://github.com/OpenIndiana/oi-userland/pull/6385/commits/52c218d648275b2902d573f4ceff3ff3d69f5d35

You can find below an example:

https://github.com/OpenIndiana/oi-userland/pull/6383/files#diff-209fdbc0b084113227a95fcc32b41372dad2eb9065fc16cffa64816f92e66cdfR37

In this case metadata is generated differently for Python 3.9.

One could imagine generating these annotations automatically in the generic manifest with n-way comparisons.

Aurélien

**Comment by balrog-nona** [comment]
Thank you very much for your explanations. I'll pay attention to this when I do Python related components.

**Comment by mtelka** [comment]
The `generic-manifest.p5m` files are no longer needed and support for them was already [removed](https://github.com/OpenIndiana/oi-userland/pull/14668).
